### PR TITLE
[#1879] Remove unnecessary test mode logic in ConfigSystemTest

### LINKED ISSUE #1879: Mismatch between system test case and actual RepoSense behavior
**Tell us about your environment**

* **RepoSense Version:** master branch
* **OS and Version:** Windows 10
* **Web Browser and Version (if applicable):** N.A.

**Please include the steps to reproduce the bug.**

1. run `gradlew run -Dargs="-c src/systemTest/resources/ConfigSystemTest -f java adoc -u 1/11/2017 -F"`

The above is to simulate the `test30DaysFromUntilDateWithFindPreviousAuthors` config test case but without the test mode flag.

**What was expected to happen?**
Both commit and authorship json files produced are identical

**What actually happened? Please include a screenshot of the output.**
The 'expected' json file contains commits from the author 'nbriannl' which are not present in the actual normal reposense run. 

This might be related to the test mode flag causing the following flag to be set `AuthorConfiguration.hasAuthorConfigFile = false;` in test mode which causes new authors to be added to the analysis when the intended behavior would be to not include any new authors for that repo. 

**If possible, include the URL to your RepoSense report or log files (if any).**

### PR DESCRIPTION
**Pull request by sikai00**
Fixes #1879 
Possibly part of #1873 

### Proposed commit message

```
Due to legacy reasons, the test mode flag was added in order to allow 
systemtest to pass. However, systemtest was previously failing because 
the json file were incorrect and should not have included the addition 
of authors not listed in author config files.

In reality, RepoSense's behavior is correct and thus do not require 
hasAuthorConfiguration to be set to false for all the systemtest.

Let's update the json files to more accurately match the actual 
behavior of RepoSense. Let's also remove the setting of 
hasAuthorConfiguration to false if test mode is enabled.
```

### Other information

Should only be merged after #1910.

This is currently failing since LocalConfigSystemTest does not have author-config.csv, however due to the global variable RepoSense believes it does.

### CODE REVIEW

**Comment by dcshzj** [comment]
Are you able to adjust the PR title to be more of an action? The title should describe what the PR will do/change instead of what the issue was. Thanks!

**Comment by sikai00** [comment]
> Are you able to adjust the PR title to be more of an action? The title should describe what the PR will do/change instead of what the issue was. Thanks!

I have updated the title as per your request, I believe this should more accurately reflect what this PR is doing!

**Review by ckcherry23 — CHANGES_REQUESTED** [review]
Hi @sikai00, I think there are some conflicts that need to be resolved as the structure of `commits.json` files had to be modified in #1882.

**Review by ckcherry23 — APPROVED** [review]
LGTM!

File: docs/dg/cli.md [hunk 1]
```diff
@@ -18,9 +18,6 @@ syntax reference under Appendix section of User Guide.
 <box type="info" seamless>
 
 * Used in `ConfigSystemTest`.
-* Some test cases requires `AuthorConfiguration.hasAuthorConfigFile` to be set to `false` to pass. The exact
-reason is unknown as the test cases are pretty old to track back. Tracing this shows that there is something to do
-with `AnnotatorAnalyzer.findAuthorInLine()` where `AuthorConfiguration.hasAuthorConfigFile()` is called.
```
Comment (chan-j-d) [inline docs/dg/cli.md:23]: Great catch for this, I actually forgot this statement was there.

**Review by chan-j-d — APPROVED** [review]
LGTM!

**Comment by chan-j-d** [comment]
On a tangentially related note, I think we should create an issue regarding the test mode flag, to see if there's any other uses for it and otherwise we should consider removing it

**Review by yhtMinceraft1010X — APPROVED** [review]
LGTM

**Comment by dcshzj** [comment]
@sikai00 Are you able to resolve the file conflicts? Thanks!

**Comment by sikai00** [comment]
> @sikai00 Are you able to resolve the file conflicts? Thanks!

I have merged and resolved the conflict. Thanks for letting me know!

**Pull request merged**